src/pages/constitution/Constitutions.tsx

```diff
@@ -0,0 +1,96 @@
+import React, { useEffect, useState } from 'react';
+import {
+  IonContent,
+  IonHeader,
+  IonPage,
+  IonToolbar,
+  IonList,
+  IonItem,
+  IonLabel,
+  IonTitle,
+} from '@ionic/react';
+import constitution from "../../assets/data/constitution.json";
+import './Constitution.css';
+
+const Constitutions: React.FC = () => {
+  const [searchTerm] = useState("")
+  const [searchableProvisions, setSearchableProvisions] = useState<any>([]);
+  const [currentSegement] = useState("all");
+
+  useEffect(() => {
+    const searchData = setupConstitutionSearch();
+    setSearchableProvisions(searchData);
+  }, [])
+
+  const setupConstitutionSearch = () => {
+    let searchData: { titleLower: string; title: string | null; content: string; id: string; }[] = [];
+    // everything that can contain searchable text
+    const selector = '.akn-p, .akn-listIntroduction, .akn-intro, .akn-wrapUp';
+    const body = new DOMParser().parseFromString(constitution.body, 'text/html');
+
+    body.querySelectorAll('.akn-section').forEach((section) => {
+      // gather text content
+      const text: (string | null)[] = [];
+      section.querySelectorAll(selector).forEach(elem => {
+        text.push(elem.textContent);
+      });
+      let titleSelector = section.querySelector('h3');
+      let title = titleSelector ? titleSelector.textContent : "";
+
+      searchData.push({
+        titleLower: title ? title.toLocaleLowerCase() : "",
+        title: title,
+        content: text.join(' ').toLocaleLowerCase(),
+        id: section.id
+      });
+
+    });
+
+    return searchData;
+  }
+
+  const renderSearchResults = () => {
+    const needle = searchTerm.toLocaleLowerCase();
+
+    const provisions = searchableProvisions.filter((x: any) => x.titleLower.includes(needle) || x.content.includes(needle));
+
+    return <>
+      <IonList>
+        {renderProvisions(provisions)}
```

Comment:
Here's how we have done something very similar using Ionic and Vue (the principle is the same in React).

https://github.com/laws-africa/bylaws-ionic-demo/blob/master/src/views/Work.vue#L11 is where we show the TOC component.

https://github.com/laws-africa/bylaws-ionic-demo/blob/master/src/components/TocMenu.vue is where we define that component and iterate over the top-level items. https://github.com/laws-africa/bylaws-ionic-demo/blob/master/src/components/TocItem.vue is where we recursively handle the children.

It doesn't have to be exactly like that, but the principle is sound.